services/system/AuthDelegate/src/AuthDelegate.cpp

```diff
@@ -78,6 +78,13 @@ namespace SystemService
 
    void AuthDelegate::setOwner(psibase::AccountNumber owner)
    {
+      auto table  = db.open<AuthDelegateTable>();
+      auto record = table.getIndex<0>().get(owner);
+      if (record.has_value())
+      {
+         check(record->owner != getSender(), "circular ownership");
```

Comment:
What about a cycle of three or more accounts?

Reply:
Then they are bricked.
We had nothing before, so this is just intended to be a slight improvement for catching the simplest invalid scenario.